metals/src/main/scala/scala/meta/internal/metals/clients/language/ConfiguredLanguageClient.scala

```diff
@@ -37,13 +37,24 @@ final class ConfiguredLanguageClient(
   }
 
   override def metalsStatus(params: MetalsStatusParams): Unit = {
-    if (clientConfig.statusBarState == StatusBarState.On) {
+    val level =
+      params.level match {
+        case "error" => MessageType.Error
+        case "warn" => MessageType.Warning
+        case _ => MessageType.Info
+      }
+    val statusBarState =
+      params.getStatusType match {
+        case StatusType.bsp => clientConfig.bspStatusBarState()
+        case _ => clientConfig.statusBarState()
+      }
+    if (statusBarState == StatusBarState.On) {
       underlying.metalsStatus(params)
-    } else if (params.text.nonEmpty && !pendingShowMessage.get()) {
-      if (clientConfig.statusBarState == StatusBarState.ShowMessage) {
-        underlying.showMessage(new MessageParams(MessageType.Log, params.text))
-      } else if (clientConfig.statusBarState == StatusBarState.LogMessage) {
-        underlying.logMessage(new MessageParams(MessageType.Log, params.text))
+    } else if (params.logMessage.nonEmpty && !pendingShowMessage.get()) {
+      if (statusBarState == StatusBarState.ShowMessage) {
+        underlying.showMessage(new MessageParams(level, params.logMessage))
```

Comment:
Potentially if we have a command defined, we could do `showMessageRequest` instead so that users would be able to use the same command in other editora

Reply:
The drawback is that I added a new client command, so if the client doesn't implement it yet, pressing `Reconnect.` will confusingly do nothing.

Reply:
We could just use ` ServerCommands.BuildConnect` command directly. If that is added to the capabilities then the editor should now to redirect the command to the server

Reply:
okay, I did that